pkg/webhook/cluster/validating_webhook.go

```diff
@@ -292,3 +291,74 @@ func (a *ManagedClusterValidatingAdmissionHook) allowUpdateClusterSet(userInfo a
 	status.Allowed = true
 	return status
 }
+
+// allowUpdateLabel use "managedclusters/label" permission checks whether a request user has been authorized to add/remove the clustersetLabel to/from ManagedCluster
+func (a *ManagedClusterValidatingAdmissionHook) allowUpdateLabel(userInfo authenticationv1.UserInfo, resourceName string) *admissionv1beta1.AdmissionResponse {
+	status := &admissionv1beta1.AdmissionResponse{}
+
+	extra := make(map[string]authorizationv1.ExtraValue)
+	for k, v := range userInfo.Extra {
+		extra[k] = authorizationv1.ExtraValue(v)
+	}
+
+	sar := &authorizationv1.SubjectAccessReview{
+		Spec: authorizationv1.SubjectAccessReviewSpec{
+			User:   userInfo.Username,
+			UID:    userInfo.UID,
+			Groups: userInfo.Groups,
+			Extra:  extra,
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Group:       "cluster.open-cluster-management.io",
+				Resource:    "managedclusters",
+				Subresource: "label",
+				Name:        resourceName,
+				Verb:        "create",
+			},
+		},
+	}
+	sar, err := a.kubeClient.AuthorizationV1().SubjectAccessReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		status.Allowed = false
+		status.Result = &metav1.Status{
+			Status: metav1.StatusFailure, Code: http.StatusForbidden, Reason: metav1.StatusReasonForbidden,
+			Message: err.Error(),
+		}
+		return status
+	}
+
+	if !sar.Status.Allowed {
+		status.Allowed = false
+		status.Result = &metav1.Status{
+			Status: metav1.StatusFailure, Code: http.StatusForbidden, Reason: metav1.StatusReasonForbidden,
+			Message: fmt.Sprintf("user %q cannot add/remove a label %q to/from ManagedClusters", userInfo.Username, resourceName),
+		}
+		return status
+	}
+	status.Allowed = true
+	return status
+}
+
+// allowUpdateClusterSet check where a request user has been authorized to add/remove the clustersetLabel to/from ManagedCluster
+func (a *ManagedClusterValidatingAdmissionHook) allowUpdateClusterSet(userInfo authenticationv1.UserInfo, clusterSetName string) *admissionv1beta1.AdmissionResponse {
+	// check "managedclustersets/join" permission
+	if status := a.allowJoinClusterSet(userInfo, clusterSetName); status.Allowed {
+		return status
+	}
+	// check "managedclusters/label" permission with resource name "labelKey:labelValue"
+	if status := a.allowUpdateLabel(userInfo, clusterSetLabel+":"+clusterSetName); status.Allowed {
```

Comment:
to my memory, when check join we also check both if user can join old  and new clusterset.

Reply:
Yes, we did the check in https://github.com/open-cluster-management-io/registration/pull/203/files#diff-042b72f0f7d4c234eb9f079b0fdfa71967b9f9c97dc0a121084177114ac3d1f6L226

Reply:
To my understanding, you should have a check func that look at the changed labels on the cluster, filter out the labels that is not used to do rbac check, then using the remaining labels to do rbac check